File: backend/src/models/Subscription.js
const mongoose = require('mongoose');

const subscriptionSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  plan: {
    type: String,
    required: true,
    enum: ['basic', 'premium', 'family', 'keto', 'vegetarian']
  },
  status: {
    type: String,
    enum: ['active', 'paused', 'cancelled'],
    default: 'active'
  },
  deliveryDays: [{
    type: String,
    enum: ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN']
  }],
  deliveryTime: {
    type: String,
    default: '14:00' // Default delivery time in 24-hour format
  },
  startDate: {
    type: Date,
    default: Date.now
  },
  endDate: {
    type: Date
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

// Update timestamp on save
subscriptionSchema.pre('save', function(next) {
  this.updatedAt = Date.now();
  next();
});

const Subscription = mongoose.model('Subscription', subscriptionSchema);

module.exports = Subscription;

File: frontend/lib/models/subscription.dart
class Subscription {
  final String id;
  final String userId;
  final String plan;
  final String status;
  final List<String> deliveryDays;
  final String deliveryTime;
  final DateTime startDate;

  Subscription({
    required this.id,
    required this.userId,
    required this.plan,
    required this.status,
    required this.deliveryDays,
    required this.deliveryTime,
    required this.startDate,
  });

  factory Subscription.fromJson(Map<String, dynamic> json) {
    return Subscription(
      id: json['_id'] as String,
      userId: json['userId'] as String,
      plan: json['plan'] as String,
      status: json['status'] as String,
      deliveryDays: List<String>.from(
        json['deliveryDays'].map((day) => day as String),
      ),
      deliveryTime: json['deliveryTime'] as String,
      startDate: DateTime.parse(json['startDate'] as String),
    );
  }
}
